### ISSUE TITLE
Update the test suite to not use the deprecated assertion methods

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5771]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)| @Julian reported|
|-|-|
|Trac ID|trac#5771|
|Type|enhancement|
|Created|2012-07-12 02:23:17Z|

Following up on [#4990](https://github.com/twisted/twisted/issues/4990), the test suite should use the assert* style methods.

This ticket is a preliminary requirement for merging [#4991](https://github.com/twisted/twisted/issues/4991).

Attachments:

* [deprecate-fail-fix-suite.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/822/8229fd8e613db20759bb450619baee864c97ead4/e5ce1d110685a608ee9525c2e77a95aef3dfabb7.patch) (438942 bytes) - added by Julian on 2012-07-12 02:23:37Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5771 5771
type__enhancement enhancement
reporter__Julian Julian
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__trial trial
keywords__ 
time__1342059797000000 1342059797000000
changetime__1388589659000000 1388589659000000
version__None None
owner__Julian Julian
cc__jml cc__Julian
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @Julian|
|-|-|

Thanks for working on this issue.  The maximum patch length for review is about 800 lines, whereas this one comes in at over 10,000 lines.  I suggest splitting the work up by sub-project (and for core, splitting it up even more than that).  Then the patch can be addressed piece by piece, and no one has to try to read and accept over ten thousand lines of changes all at once.  Each patch should have its own ticket.

Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_4"></a>|@jyknight commented|
|-|-|

Meh.

It's only 3324 lines if you eliminate all the context lines. And, bonus, it looks easier to review that way too (seriously, not even joking; don't need context to see that self.failUnless(...) -> self.assertTrue(...) is right.)

I expect the changes to trial itself to be an issue (may want to continue using deprecated methods in tests for the deprecated methods, for example), so certainly that set of changes should be examined with more care. But the rest looks pretty darn obvious...

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Replying to [jknight](#note_4):
> It's only 3324 lines if you eliminate all the context lines. And, bonus, it looks easier to review that way too (seriously, not even joking; don't need context to see that self.failUnless(...) -> self.assertTrue(...) is right.)
> 
> I expect the changes to trial itself to be an issue (may want to continue using deprecated methods in tests for the deprecated methods, for example), so certainly that set of changes should be examined with more care. But the rest looks pretty darn obvious...

Do you want to do the code review on it, then?

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_6"></a>|@Julian set status to `closed`|
|-|-|

This is being broken up and done in tickets [#6220](https://github.com/twisted/twisted/issues/6220) - [#6233](https://github.com/twisted/twisted/issues/6233).

Closing as a duplicate since there isn't really anything useful here.

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|julian commented|
|-|-|

(In [41086]) Merge remove-deprecated-test-methods-names-6222: Remove to-be-deprecated test methods from twisted.names

Author: Julian
Reviewers: tom.prince, exarkun
Fixes: [#6222](https://github.com/twisted/twisted/issues/6222)

Also removes the few from twisted.python. See [#5771](https://github.com/twisted/twisted/issues/5771) for more info.
